### DISCUSSION TITLE
Product Subscription

### DISCUSSION THREAD

**Discussion by denerFernandes** [vendurehq/vendure#1118] [General]
Hi Guys, 

I have a demand to developer a subscription products support (with recurring payments) my idea is:

- Create an option on the product to mark the product as a subscription
- If is subscription product show a list with subscription plans  to select available plans for this product
- Create a list with all subscription with a filters (Active, Cancelled, ...)
- Create a page to show subscription details (Display costumer data, payments, items shipped) and options to change status (approve, cancel)
- Create a page to manage subscriptions plans
- Integrate this with payments methods (For now Stripe)

1. Vendure does not have anything like this correct?
2. Would it be of interest to the community?
3. Any suggestion?

Thanks guys.

**Comment by michaelbromley** [comment]
Hi,

This sounds like a reasonable high-level plan of how to implement.

1. No, nothing like that built in. You'd need to built it yourself as a plugin.
2. Probably! If you build it with possible open-sourcing in mind, and keep the implementation generic & decoupled then you can always release it later once you have something working and well-tested.

**Reply by denerFernandes** [reply]
Thanks for your feedback, when i have something to show I would appreciate your feedback again.

**Reply by badrange** [reply]
I just started following this discussion because I'm looking into the same. How has your work progressed @denerFernandes ?
